language/move-prover/boogie-backend/src/spec_translator.rs

```diff
@@ -538,26 +681,62 @@ impl<'env> SpecTranslator<'env> {
             Operation::Trace => self.trace_value(node_id, TraceItem::Explicit, || {
                 self.translate_exp(&args[0])
             }),
-            Operation::MaxU8 => emit!(self.writer, "$Integer($MAX_U8)"),
-            Operation::MaxU64 => emit!(self.writer, "$Integer($MAX_U64)"),
-            Operation::MaxU128 => emit!(self.writer, "$Integer($MAX_U128)"),
-            Operation::AbortCode | Operation::AbortFlag => unimplemented!(),
+            Operation::MaxU8 => emit!(self.writer, "$MAX_U8"),
+            Operation::MaxU64 => emit!(self.writer, "$MAX_U64"),
+            Operation::MaxU128 => emit!(self.writer, "$MAX_U128"),
+            Operation::WellFormed => self.translate_well_formed(&args[0]),
+            Operation::AbortCode | Operation::AbortFlag => {
+                unimplemented!()
+            }
             Operation::NoOp => { /* do nothing. */ }
         }
     }
 
+    fn translate_check_event_store(&self, args: &[Exp]) {
```

Comment:
@junkil-park The event store operations have now become internal operations in the Exp AST. Here we translate them to Boogie. There is another place where we generate them where I also mention you.